### PR TITLE
Experience/11334/redesign release notes

### PR DESCRIPTION
**Pull request by etanb**
This PR adds a newly stylized `/product/release-notes` page. We're leaving the old `release-notes.md` in place for now and will cleanse them all once we make all page updates.

### CODE REVIEW

**Review by jpandersen87 — APPROVED** [review]

File: frontend-react/src/content/product/sidenav.mdx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<SideNav items={[
+    <SideNavItem href="/product/overview">Getting started</SideNavItem>,
```
Comment (penny-lischer) [inline frontend-react/src/content/product/sidenav.mdx:2]: Is think correct? should it be` /getting-started` or should the label be "Overview of ReportStream"?

**Comment by KeithNava** [comment]
@etanb I was hoping we could use the `webx-sprint-77-feature-branch` as the release branch for this work. Can you update the base branch for this PR?

File: frontend-react/src/content/product/sidenav.mdx (hunk 1 above)
Comment (etanb) [reply frontend-react/src/content/product/sidenav.mdx:2]: just following the current sidenav hierarchy we have on the site. we can update as we add pages if that works for you?

**Review by penny-lischer — APPROVED** [review]
lgtm

**Review by yudonoodles — APPROVED** [review]
LGTM!

File: frontend-react/src/AppRouter.tsx [hunk 2]
```diff
@@ -248,6 +248,14 @@ export const appRoutes: RouteObject[] = [
                     },
                 ],
             },
+            {
+                path: "/product/release-notes",
```
Comment (penny-lischer) [inline frontend-react/src/AppRouter.tsx:252]: Should this path be content/about/release-notes?

Comment (etanb) [reply frontend-react/src/AppRouter.tsx:252]: Following the pattern of current route: https://reportstream.cdc.gov/product/release-notes

**Pull request merged**